model/riscv_vmem.sail

```diff
@@ -162,9 +166,16 @@ function pt_walk(sv_params,
                   let pa  = (ppn << pagesize_bits) | zero_extend(offset_of_va(va));
                   PTW_Success(pa, pte, pte_addr, level, global', ext_ptw)
                 }
-              }
-              else {
-                let pa = (ppns << pagesize_bits) | zero_extend(offset_of_va(va));
+              } else if (haveSvnapot() & ext_pte_flags[N] == 0b1 & ppns[3 .. 0] != 0b1000) then {
+                  // Only 64K NAPOT PTEs are valid
+                  PTW_Failure(PTW_Invalid_PTE(), ext_ptw)
+              } else {
+                let final_ppn : bits(64) = match (haveSvnapot(), ext_pte_flags[N]) {
```

Comment:
I really think an `if` is clearer for these sorts of things. Also you can just reuse the `ppns` name. And use `@` instead of `append`.

```
let ppns = if haveSvnapot() & ext_pte_flags[N] == 0b1
  then ppns[63 .. 4] @ vpn_j[3 .. 0]
  else ppns;
```

Reply:
updated.